### PR TITLE
Add Workload Identity as an auth mechanism

### PR DESCRIPTION
**Pull request by rajashekhargundeti**

### CODE REVIEW

File: GettingStarted.md [hunk 1]
```diff
@@ -73,6 +75,15 @@ kubectl create secret generic oci-config \
...
 Instance principal would work only on OKE cluster.
 Access should be granted using Access Policies(See [Access Policies](#access-polices) section).
+
+<a name="auth-workload-identity"></a>
+### Workload Identity
+Workload Identity works only in OKE Enhanced clusters.
+
+Access should be granted using Access Policies(See [Access Policies for Workloads](#access-policies-workloads) section).
+
+Worklaod Identity uses a Resource Principal auth, which requires settings a couple ENV variables on the pod, including the region where the cluster is deployed. To achieve this, make sure to specify the `provider.oci.auth.types.workloadIdentity.enabled=true` and `provider.oci.auth.types.workloadIdentity.region=<region>` parameters in the `values.yaml` for the Helm chart deployment, or as an inline parameters.
```
Comment (rajashekhargundeti) [inline GettingStarted.md:85]: type: workload
please double check whether this is provider.oci.auth.types.workload or provider.oci.auth.types.workloadIdentity

File: internal/server/server.go [hunk 2]
```diff
@@ -195,6 +226,51 @@ func parseAuthConfig(secret *core.Secret, authConfigSecretName string) (*types.A
...
+		return "", fmt.Errorf("unable to get k8s client: %v", err)
+	}
+	ttl := int64((15 * time.Minute).Seconds())
+	resp, err := clientSet.CoreV1().
+		ServiceAccounts(podInfo.Namespace).
+		CreateToken(context.Background(), podInfo.ServiceAccountName,
+			&authenticationv1.TokenRequest{
+				Spec: authenticationv1.TokenRequestSpec{
+					ExpirationSeconds: &ttl,
+					Audiences:         []string{"oci", "api"},
```
Comment (rajashekhargundeti) [inline internal/server/server.go:255]: we may leave Audiences empty

**Comment by rajashekhargundeti** [comment]
This objective isn't implemented:
The provider should accept audience parameter for token generation or to capture a particular token with correct audience from the provided service account token

**Review by rajashekhargundeti — COMMENTED** [review]
This objective isn't implemented:
The provider should accept audience parameter for token generation or to capture a particular token with correct audience from the provided service account token

File: GettingStarted.md (hunk 1 above)
Comment (balkrishna93) [reply GettingStarted.md]: Updated the details to workload

File: internal/server/server.go (hunk 2 above)
Comment (balkrishna93) [reply internal/server/server.go]: Updated.

**Review by YashwantGohokar — APPROVED** [review]

**Review by balkrishna93 — APPROVED** [review]

**Comment by balkrishna93** [comment]
Closing the pull request in favor of https://github.com/oracle/oci-secrets-store-csi-driver-provider/pull/32

**Pull request closed without merging**